pkg/gofr/datasource/mongo/mongo.go

```diff
@@ -108,18 +119,31 @@ func (c *Client) Connect() {
 	}
 
 	if err = m.Ping(ctx, nil); err != nil {
-		c.logger.Errorf("could not connect to mongoDB at %v due to err: %v", c.config.URI, err)
+		c.logger.Errorf("could not connect to mongoDB at %v due to err: %v", host, err)
 		return
 	}
 
-	c.logger.Logf("connected to mongoDB successfully at %v to database %v", c.config.URI, c.config.Database)
+	c.logger.Logf("connected to mongoDB successfully at %v to database %v", host, c.config.Database)
 
 	mongoBuckets := []float64{.05, .075, .1, .125, .15, .2, .3, .5, .75, 1, 2, 3, 4, 5, 7.5, 10}
 	c.metrics.NewHistogram("app_mongo_stats", "Response time of MONGO queries in milliseconds.", mongoBuckets...)
 
 	c.Database = m.Database(c.config.Database)
 
-	c.logger.Logf("connected to MongoDB at %v to database %v", uri, c.Database)
+	c.logger.Logf("connected to MongoDB at %v to database %v", host, c.Database)
+}
+
+func getDBHost(uri string) (host string) {
+	// regular expression to extract the host
+	re := regexp.MustCompile(`mongodb://.*?:.*?@(.*?):\d+/.*`)
+
+	matches := re.FindStringSubmatch(uri)
+
+	if len(matches) > 1 {
+		host = matches[1]
+	}
+
+	return host
```

Comment:
Regexp are evil, and also they are slow

Here you are parsing something that look a like an URI

I mean it's `scheme://user:pass@host:port/path`

Can't you use net/url package ? 

https://pkg.go.dev/net/url#ParseRequestURI

Then Hostname https://pkg.go.dev/net/url#URL.Hostname